metadata/ota_metadata/metadata_gen.py

```diff
@@ -191,15 +191,57 @@ def gen_metadata(
     target_abs = Path(os.path.abspath(target_dir))
     ignore = ignore_rules(target_dir, ignore_file)
 
+    # If ignore file has the following directories:
+    #    1, "home/autoware/*/build"
+    #    2, "home/autoware/*/src"
+    # We will NOT simply ignore files under them.
+    # We will check the following:
+    #    1, If the file is set as the target of a symblink ?
+    #    2, If the file falls in some special folder pattern ?
+    # Why we need to do this? It is explained in this document
+    # https://tier4.atlassian.net/wiki/x/JoC21Q
+    check_patterns = [r"home/autoware/[^/]+/build", r"home/autoware/[^/]+/src"]
```

Comment:
You can pre-compile the regex pattern here to improve the efficiency, like
```
check_patterns = [re.compile(r"home/autoware/[^/]+/build"), re.compile(r"home/autoware/[^/]+/src")]
```

`re.compile` will return the `Pattern` object, then you can use `Pattern` object's `search` method to check the line in ignores.txt as follow:
```
pattern.search(str(rule))
```